mafiasi/base/models.py

```diff
@@ -47,13 +39,18 @@ def __str__(self):
         return self.name
 
 
-class PasswdEntry(models.Model):
-    username = models.CharField(max_length=40, unique=True)
-    full_name = models.CharField(max_length=60)
-    gid = models.BigIntegerField()
-
-    def __str__(self):
-        return self.username
+@receiver(pre_save, sender=Yeargroup)
+def yeargroup_pre_save(sender, instance, **kwargs):
+    if instance.id is None:
+        # when instance.id is None, the object was just created
+        try:
+            # Sometimes, gids are reused. In this case, unset the old group's gid first.
+            same_gid_group = Yeargroup.objects.get(gid=instance.gid)
+            same_gid_group.gid = None
+            same_gid_group.save()
```

Comment:
Currently, the gid is still used in `mafiasi/jabber/models.py:254` because I assume that student yeargroups have a gid and employee/external yeargroups don't. However, this is only necessary to determine if a `j` has to be added to the group. In my opinion, the cleanest option would be to change the yeargroup slugs/names to e.g. `j2021` instead of `2021`. This would require a little manual change in the current setup, but would make conversions to/from IRZ and Jabber unnecessary, since both of them already use the `j2021` form. What do you think?

Reply:
I agree, changing the yeargroup names to be consistent with the IRZ would probably be the cleanest solution. I'm not sure whether we want to apply that to the existing yeargroups and rename them as well – maybe some clients get confused from that. Though I guess that should not be a real problem and fixed when the Jabber clients start a new session.

Regardless of that, we should be able to robustly decide whether to add a `j` based on the format of the slug (i.e., 4 digits or something else?), without relying on the gid?

Reply:
The yeargroups in Jabber also contain the leading `j` (c.f. `mafiasi/jabber/models.py:256`). Therefore, no problems would occur for Jabber clients. This only affects the internal representation in the dashboard which is only used to create the Jabber group and map the user correctly.
So, if we always keep the `j`, we would never have to decide when to add it.

Reply:
Ok, then we should definitely switch to keeping the `j` for the existing yeargroups as well.

Reply:
I suggested an implementation in 9445b3f98ede0ea64fca3a084df6780efebf32ef, maybe that makes clearer what I mean.

Reply:
Looks good.